spec/filters/geoip_spec.rb

```diff
@@ -221,6 +221,23 @@
         subject
       end
     end
+  end
+
+  describe "returned object identities" do
+    let(:plugin) { LogStash::Filters::GeoIP.new("source" => "message") }
+    let(:geo_data) { plugin.get_geo_data_for_ip("8.8.8.8") }
+
+    before do
+      plugin.register
+    end
+
+    it "should dup the objects" do
+      event = {}
+      alt_event = {}
+      plugin.apply_geodata(geo_data, event)
+      plugin.apply_geodata(geo_data, alt_event)
 
+      event["geoip"].each {|k,v| expect(v).not_to eql(alt_event[k]) }
```

Comment:
I was gonna ask that, but then I remembered that Ruby has like 6 ways to measure equality and maybe rspec's eql used the one that compares object id or something and then my mind wandered onto more pleasant things like cheese and hockey.